median/plugin.go

```diff
@@ -38,13 +42,24 @@ func (p *Plugin) NewMedianFactory(ctx context.Context, provider types.MedianProv
 			}
 		}),
 		OnchainConfigCodec: provider.OnchainConfigCodec(),
-		ReportCodec:        provider.ReportCodec(),
 	}
+
 	if cr := provider.ChainReader(); cr != nil {
-		factory.ContractTransmitter = &chainReaderContract{cr, types.BoundContract{Name: "median"}}
+		factory.ContractTransmitter = &chainReaderContract{chainReader: cr}
 	} else {
 		factory.ContractTransmitter = provider.MedianContract()
 	}
+
+	if codec := provider.Codec(); codec != nil {
+		var err error
+		if factory.ReportCodec, err = newReportCodec(codec); err != nil {
+			return nil, err
+		}
+	} else {
+		lggr.Warn("No codec provided, defaulting back to median specific ReportCodec")
```

Comment:
Not sure if we should log warn for default behaviour, maybe info?

Reply:
or even always set it to reportCodec and overwrite it if codec is provided and log that we are using codec instead of reportcodec

Reply:
I copied what was done for the chain reader.  Happy to make it debug for both since we aren't expecting people to use it.

Reply:
I think changing the level here is important. We don't need to be confusing node ops by "warning" them about the expected behavior.